plugins/modules/docker_swarm.py

```diff
@@ -569,7 +584,8 @@ def join(self):
                 self.client.join_swarm(
                     remote_addrs=self.parameters.remote_addrs, join_token=self.parameters.join_token,
                     listen_addr=self.parameters.listen_addr, advertise_addr=self.parameters.advertise_addr,
-                    data_path_addr=self.parameters.data_path_addr)
+                    data_path_addr=self.parameters.data_path_addr,
+                    data_path_port=self.parameters.data_path_port)
```

Comment:
Same here. You need to create a `kwargs = {}`, fill it with `data_path_port` if that's not `None`, and pass `**kwargs` here to avoid problems with older Docker SDK for Python versions.

Reply:
Discarded that as after more digging, data_path_port can only be used when initializing a swarm.